suite/spdz/src/test/java/dk/alexandra/fresco/suite/spdz/TestFakeTripGen.java

```diff
@@ -49,4 +53,119 @@ public void testInputMasks() {
+          Assert.assertNotNull(masks[towardsPlayer-1].getRealValue());
+        }
+    }
+	
+	@Test
+    public void testExpPipe() {
+        int amount = 2;
+        int noOfParties = 2;
+        BigInteger modulus = new BigInteger("6703903964971298549787012499123814115273848577471136527425966013026501536706464354255445443244279389455058889493431223951165286470575994074291745908195329");
+        BigInteger alpha = new BigInteger("5081587041441179438932635098620319894716368628029284292880408086703438041331200877980213770035569812296677935118715454650749402237663859711459266577679205");
+        List<SpdzSInt[][]> expPipes = FakeTripGen.generateExpPipes(amount, noOfParties, modulus, alpha);        
+        for(SpdzSInt[][] pipe : expPipes) {
+          SpdzSInt[] as = pipe[0];
+          SpdzSInt[] bs = pipe[1];
+          for(int i = 0; i < as.length; i++) {
+            BigInteger share = as[i].value.getShare().add(bs[i].value.getShare()).mod(modulus);
+            BigInteger mac = as[i].value.getMac().add(bs[i].value.getMac()).mod(modulus);
+            Assert.assertEquals(BigInteger.ZERO, mac.subtract(share.multiply(alpha).mod(modulus)));       
+          }
+        }
+    }
+	
+	@Test
+	public void testBitGen() {
+	  int amount = 100000;
+	  int noOfParties = 2;
+	  BigInteger modulus = new BigInteger("6703903964971298549787012499123814115273848577471136527425966013026501536706464354255445443244279389455058889493431223951165286470575994074291745908195329");
+	  BigInteger alpha = new BigInteger("5081587041441179438932635098620319894716368628029284292880408086703438041331200877980213770035569812296677935118715454650749402237663859711459266577679205");
+	  List<SpdzSInt[]> bits = FakeTripGen.generateBits(amount, noOfParties, modulus, alpha);
+	  for(SpdzSInt[] b : bits) {
+	    BigInteger val = b[0].value.getShare().add(b[1].value.getShare()).mod(modulus);
+	    BigInteger mac = b[0].value.getMac().add(b[1].value.getMac()).mod(modulus); 
+	    
+	    Assert.assertTrue(val.equals(BigInteger.ZERO) || val.equals(BigInteger.ONE));
+	    Assert.assertEquals(BigInteger.ZERO, mac.subtract(val.multiply(alpha).mod(modulus)));	    
+	  }
+	}
+	
+	@Test
+	public void testElementToBytes() {	  
+	  SpdzElement element = new SpdzElement(BigInteger.valueOf(200), BigInteger.ONE, BigInteger.ZERO);
+	  FakeTripGen.elementToBytes(element, 1);
+	  try {
+	    FakeTripGen.elementToBytes(element, 0);
+	    Assert.fail("Should have cast an exception");
+	  } catch(MPCException e) {
+	    
+	  }
+	  
+	  element = new SpdzElement(BigInteger.ONE, BigInteger.valueOf(200), BigInteger.ZERO);
+      FakeTripGen.elementToBytes(element, 1);
+      try {
+        FakeTripGen.elementToBytes(element, 0);
+        Assert.fail("Should have cast an exception");
+      } catch(MPCException e) {
+        
+      }
+	}
+	
+	@Test
+	public void testBigIntToBytes() {
+	  BigInteger b = BigInteger.valueOf(200);
+	  int size = 1;
+	  FakeTripGen.bigIntToBytes(b, size);
```

Comment:
Again, the result does not seem to be used. Shouldn't we test that the method actually gives the byte representation of `b`?